### PR TITLE
Reactive forms for dpDayPicker directive

### LINKED ISSUE #76: New version with support for material as show in demo app
Exist any changes that you publish new version for integrate with material as show the demo in master branch using DatePickerDirective?, actually i used the last version published in npm 1.3.0.
Thanks.

### PR DESCRIPTION
**Pull request by isaacplmann**
Fixes #76

### CODE REVIEW

**Comment by isaacplmann** [comment]
I also wrote some e2e tests for the reactive forms support, but it requires pulling in the ReactiveFormsModule into the demo.  Should I include those tests in the PR?

**Comment by vlio20** [comment]
Sure, don't see why not. I will go over the PR, in the next few hours.

File: README.md [hunk 1]
```diff
@@ -207,17 +207,24 @@ Here are the available configurations:
...
 
 i.e.
 ```html
-<input name="someName" [(ngModel)]="selectedDate" [dpDayPicker]="config"></input>
+<input name="someName" [(ngModel)]="selectedDate" [dpDayPicker]="config" />
+```
+
+or using reactive forms:
+```html
+<input name="someName" formControlName="selectedDate" [dpDayPicker]="config" />
```
Comment (vlio20) [inline README.md]: why do we need to support this as well? 
this one:  
```
<input name="someName" [formControl]="selectedDateFormControl" [dpDayPicker]="config" />
```
is the official way, right?

File: README.md [hunk 2]
```diff
@@ -232,7 +239,7 @@ or with `@angular/material`:
 ### Configuration:  
 In order to provide configurations to the date-picker you need to pass it to the `[dpDatePicker]` directive:
```
Comment (vlio20) [inline README.md]: shouldn't this sentence be updated well? (date-picker -> day-picker)

File: src/app/demo/demo/demo.component.html [hunk 3]
```diff
@@ -143,6 +144,20 @@
         </div>
       </div>
 
+      <div class="dp-picker-container" id="datePicker" *ngIf="pickerMode === 'dayDirectiveReactive'">
```
Comment (vlio20) [inline src/app/demo/demo/demo.component.html]: this id is already in use, please give other id

File: e2e/app.e2e-spec.ts [hunk 4]
```diff
@@ -47,6 +47,43 @@ describe('ng2-date-picker App', () => {
     });
   });
 
+  describe('dpDayPicker reactive directive', () => {
```
Comment (vlio20) [inline e2e/app.e2e-spec.ts]: can you pls also add tests for the validations? (required, min, max)

File: README.md (hunk 1 above)
Comment (isaacplmann) [inline README.md]: `[formControl]` is for a stand alone `FormControl` object.  `[formControlName]` is for referencing a `FormControl` that is part of a `FormGroup` object.  Both need to be supported, but it's easy to support them both.  I just inject the `NgControl` and angular takes care of figuring out where to get it from.

File: e2e/app.e2e-spec.ts (hunk 4 above)
Comment (isaacplmann) [inline e2e/app.e2e-spec.ts]: Those validations won't go through the directive at all - they'll be directly on the `ngModel`.  But I'll put in some tests to make sure that the directive doesn't interfere with those somehow.

Comment (vlio20) [inline e2e/app.e2e-spec.ts]: Another question, in a reactive form the validations set within the controller (in the FormGroup). How the custom validations, MI  and max will be set? I never did it before.

**Comment by isaacplmann** [comment]
I made the changes you requested, plus a couple changes in the e2e tests to make them pass now that the current month is May instead of April.

'should disable/enable month selection' and 'should check allow multiselect is working' were both depending on the current month being April.  Now they're depending on the current month being May. (*shrug*)  That should be fixed before June comes around, but it's a separate issue.

File: src/app/demo/demo/demo.component.ts [hunk 5]
```diff
@@ -21,17 +22,23 @@ export class DemoComponent {
...
   required: boolean = false;
   disabled: boolean = false;
   validationMinDate: Moment;
   validationMaxDate: Moment;
   placeholder: string = 'Choose a date...';
 
+  formGroup: FormGroup = new FormGroup({
+    datePicker: new FormControl(this.date, [
+      this.required ? Validators.required : () => undefined,
+      control => this.validationMinDate && this.config && moment(control.value, this.config.format).isBefore(this.validationMinDate)
```
Comment (vlio20) [inline src/app/demo/demo/demo.component.ts]: don't you think the component should export the min and max validations?

Comment (isaacplmann) [inline src/app/demo/demo/demo.component.ts]: I suppose you could have the library export the validation functions.  That would make it easier on users of the library.

Comment (vlio20) [inline src/app/demo/demo/demo.component.ts]: will add a task for that. thanks!

**Pull request merged**